src/controller/userController.js

```diff
@@ -34,22 +33,81 @@ class UserController {
         bio,
       });
       if (userCreated) {
-        const tokenCreated = await authentication.getToken(userCreated);
-        res.status(201).send({
-          message: `User ${userCreated.username} created successfully`,
-          id: userCreated.id,
-          username: userCreated.username,
-          email: userCreated.email,
+        const tokenCreated = await
+        Authentication.getToken(
+          userCreated.dataValues, process.env.reg_token_expiry
+        );
+        const waitingToAuthorize = await Authorize.create({
           token: tokenCreated,
+          userId: userCreated.id,
         });
+        if (waitingToAuthorize) {
+          const isEmailSent = await SendEmail.verifyEmail(email, tokenCreated);
+          if (isEmailSent) {
+            return res.status(200).send({
+              success: true,
+              message: `An email has been sent to your email address.
+                Please check your email to complete your registration`,
+            });
+          }
+          return res.status(500).send({
+            success: false,
+            message:
+              'Your registration could not be completed. Please try again',
+          });
+        }
       }
     } catch (error) {
-      res.status(500).send({
+      if (error.errors) {
+        return res.status(400).send({
+          success: false,
+          message: error.errors[0].message,
+        });
+      }
+      return res.status(500).send({
+        success: false,
         message: 'Internal server error',
+        myError: error.message,
+        error,
       });
     }
   }
 
```

Comment:
Hi, @Daymorelah, may I suggest simplifying this a bit by making the method do just one or max two things. At the moment, it's creating a user, creating authorization columns, sending a verification email, and has a lot of return statements. These things would definitely increase code complexity. Try to abstract them into middleware e.g `if (userCreated) {
        const tokenCreated = await authentication.getToken(userCreated);
        res.status(201).send({
          message: `User ${userCreated.username} created successfully`,
          id: userCreated.id,
          username: userCreated.username,
          email: userCreated.email,` in one controller and `if (waitingToAuthorize) {
          const isEmailSent = await SendEmail.verifyEmail(email, tokenCreated);
          if (isEmailSent) {
            return res.status(200).send({
              success: true,
              message: `An email has been sent to your email address.
                Please check your email to complete your registration`,
            });
          }` in another middleware and then add them to the routes. Other than that, the logic looks good!